src/cryptenroll/cryptenroll.c

```diff
@@ -253,7 +260,29 @@ static int parse_argv(int argc, char *argv[]) {
                         r = parse_path_argument(optarg, /* suppress_root= */ true, &arg_unlock_keyfile);
                         if (r < 0)
                                 return r;
+
+                        arg_unlock_type = UNLOCK_KEYFILE;
+                        break;
+
+                case ARG_UNLOCK_FIDO2_DEVICE: {
+                        _cleanup_free_ char *device = NULL;
+
+                        if (streq(optarg, "auto")) {
+                                if (arg_enroll_type == ENROLL_FIDO2)
```

Comment:
IIUC, `arg_enroll_type` is set by another option. For this check to work, the user needs to use *that* option first and *this* option second. If they do it in opposite order, the check will be skipped. I think this check should be moved to end of this function, after all the options have been processed and we know the final values.

Reply:
This is why the check is done in both the case for `ARG_UNLOCK_FIDO2_DEVICE` and for `ARG_FIDO2_DEVICE`. But yes I guess it would be better to do the check at once after the loop.